gravitee-apim-plugin/gravitee-apim-plugin-entrypoint/gravitee-apim-plugin-entrypoint-http-get/src/main/java/io/gravitee/plugin/entrypoint/http/get/HttpGetEntrypointConnector.java

```diff
@@ -0,0 +1,330 @@
+
+    @Override
+    public ListenerType supportedListenerType() {
+        return ListenerType.HTTP;
+    }
+
+    @Override
+    public Set<ConnectorMode> supportedModes() {
+        return SUPPORTED_MODES;
+    }
+
+    @Override
+    public int matchCriteriaCount() {
+        return 1;
+    }
+
+    @Override
+    public boolean matches(final ExecutionContext ctx) {
+        return (HttpMethod.GET == ctx.request().method());
+    }
+
+    @Override
+    public Completable handleRequest(final ExecutionContext ctx) {
+        return Completable.defer(
+            () -> {
+                String acceptHeader = ctx.request().headers().get(HttpHeaderNames.ACCEPT);
+                final String contentType = (acceptHeader == null || acceptHeader.isEmpty() || acceptHeader.equals(MediaType.WILDCARD))
+                    ? MediaType.TEXT_PLAIN
+                    : acceptHeader;
+                if (
+                    !contentType.equals(MediaType.APPLICATION_JSON) &&
+                    !contentType.equals(MediaType.APPLICATION_XML) &&
+                    !contentType.equals(MediaType.TEXT_PLAIN)
+                ) {
+                    return ctx.interruptWith(
+                        new ExecutionFailure(HttpStatusCode.BAD_REQUEST_400).message("Unsupported accept header: " + acceptHeader)
+                    );
+                }
+                ctx.setAttribute(ATTR_INTERNAL_RESPONSE_CONTENT_TYPE, contentType);
+
+                ctx.putInternalAttribute(
+                    InternalContextAttributes.ATTR_INTERNAL_MESSAGES_LIMIT_DURATION_MS,
+                    configuration.getMessagesLimitDurationMs()
+                );
+
+                int messagesLimitCount = configuration.getMessagesLimitCount();
+                if (ctx.request().parameters().containsKey(LIMIT_QUERY_PARAM)) {
+                    String limit = ctx.request().parameters().getFirst(LIMIT_QUERY_PARAM);
+                    if (limit != null && !limit.isEmpty()) {
+                        messagesLimitCount = Math.min(messagesLimitCount, Integer.parseInt(limit));
+                    }
+                }
+                ctx.putInternalAttribute(InternalContextAttributes.ATTR_INTERNAL_MESSAGES_LIMIT_COUNT, messagesLimitCount);
+
+                if (ctx.request().parameters().containsKey(CURSOR_QUERY_PARAM)) {
+                    String cursor = ctx.request().parameters().getFirst(CURSOR_QUERY_PARAM);
+                    if (cursor != null && !cursor.isEmpty()) {
+                        ctx.putInternalAttribute(InternalContextAttributes.ATTR_INTERNAL_MESSAGES_RESUME_LASTID, cursor);
+                    }
+                }
+
+                return Completable.complete();
+            }
+        );
+    }
+
+    @Override
+    public Completable handleResponse(final ExecutionContext ctx) {
+        return Completable.fromRunnable(
+            () -> {
+                String contentType = ctx.getInternalAttribute(ATTR_INTERNAL_RESPONSE_CONTENT_TYPE);
```

Comment:
Maybe a test is missing as it should fail because of `setAttribute` vs `getInternalAttribute`.